### ISSUE TITLE
[ANDROID ONLY] Short video on internal storage of an app won't load

### ISSUE THREAD

**Issue by jonathangreco** [TheWidlarzGroup/react-native-video#1311]
### Current behavior
My video won't start if it's duration is under 8/10 seconds (I've a video of 19 seconds (internal storage) that played correctly)

If I film and saveToCameraRoll a video regardless of its duration the player works "almost" correctly.

BUT 
If i wanna see a video from 
```
/data/user/0/my_app/files/my/folder/structure/medias/tpl/demonstration_video.mp4
// or
/data/user/0/my_app/files/my/folder/structure/medias/scene-0/demonstration_video.mp4
```
and it's length is < 10 seconds it won't works

On IOS it works perfectly (same video of course), with the controls set to true.

### Reproduction steps
Have a short video in internal app storage (2seconds, 4 seconds, 6 seconds) and try to play it on ANDROID.

### Expected behavior
The video must start, but it's not.

### Platform
* Android ExoPlayer

With multiple android device (at least 3 -> means that's a generic issue)

## Guess
I suspect that is related to the duration of the video & the Internal storage because this is the only discriminant I managed to identify throught my tests
Is some MP4 files cant be loaded for a reason ?

is someone can reproduce this issue ?

**Comment by jonathangreco** [comment]
@cobarx Hi, What do you need to reproduce this error ? Do you need more informations ?

Thanks in advance

**Comment by jonathangreco** [comment]
@cobarx ping.

Hi, I really need an answer to know if this bug is repeatable.

It's an issue that mess up my entire project so far with android. It's an emergency to me to make progress in the resolution of this issue.

**Comment by cobarx** [comment]
Hi @jonathangreco, thanks for pinging I will take a look tonight. I'm guessing I can use `adb` to get the path to the video. If not, please add instructions.

Do you mind posting one of those videos in case my phone generates a video that does work properly.

**Comment by jonathangreco** [comment]
Hi @cobarx ,

To the path for registering videos, you can use RNFS like this : 

```
export const DirHome = Platform.select({
  ios: `${RNFS.DocumentDirectoryPath}/my_app`,
  android: `${RNFS.DocumentDirectoryPath}/my_app`
});

```
An then register your videos in it.
I think any of short video will not work, I let you test it, in case I'm wrong.

Really thank you for dig into this

**Comment by jonathangreco** [comment]
Hi thanks for your tests.

I just want to make sure that you moved your videos into the internal storage of  your app before play it.
Because any video recorded and played while they are in CameraRoll or any place elsewhere will be played correctly.


I will test your short video into my app this morning.

**Comment by jonathangreco** [comment]
After test of your video stored into my app storage with this path: 

```
/data/user/0/com.app/files/App/Templates/Name_of_my_template/medias/tpl/demonstration_video.mp4
```
The same error: video won't load.
I will test your long video see if it's play correctly.


tested on random device we owned :

```

> Task :app:installDebug 
Installing APK 'app-debug.apk' on 'P028 - 7.0' for app:debug
Installed on 1 device.

```

EDIT :

Your long video played correctly on my app.

**Comment by jonathangreco** [comment]
@cobarx ping.
If you can confirm that you tried to play your video from the "internal storage" then I will try to install the new version 4.0 and see if it changes anything

Edit : 
With the latest version master the error is still happening. :(

**Comment by cobarx** [comment]
I haven't tested from internal storage, it's considerably more work to setup and I was trying to avoid that. I'm guessing in your app you have a Camera module setup and are recording from within your app. Apps only have access to files in their own sandbox and public folders like Movies, Download, etc. which require permission to access. In order to do a quick check, I was doing my best to keep it simple. It sounds like this is specifically related to local files, so I'll have to set it up.

Have you looked at logcat in Android Studio to see if there is anything helpful for diagnosing this? You should see some kind of error either accessing or decoding the video.

**Comment by jonathangreco** [comment]
> I haven't tested from internal storage

Well like you said, this issue is specifically related to local app storage. It's the whole point of this issue. I know it's a big setup to make it testable.

I will dig into logcat eventually. I've never done it, I need to practise a bit.

@cobarx Thanks again for your concerns. 


>  I'm guessing in your app you have a Camera module setup and are recording from within your app

Nope. It's way more complex. I've an API that provide a template (a zip folder) containing the videos. And I save those videos in my app storage. Then when I play those videos, only when the duration is short, the player won't work. And for longer videos it works.

My project is simple, I've a template that represent a short film that the user can reproduce (IMovie like), all videos I download from my API are just demonstration, the user have to reproduce the same film in his own environment. Sometime, a scene can have a short duration, that's why on android, my app currently not working.

I'm not native english speaker, if I'm not clear enouth, please tell me

**Comment by jonathangreco** [comment]
I tested it 6 days ago with "react-native-video": "https://github.com/react-native-community/react-native-video.git#master"
I still have this version actually.

Plus your short video you filmed have sound from memory. And it didn't worked either. Are you sure about this bug ?

**Comment by jonathangreco** [comment]
@cobarx ping :)

**Comment by cobarx** [comment]
That stack trace indicates there's a problem in the text & audio track selection code, so I was hoping the recent crash fix in master would fix it for you as well. 

I'll see if I can test it tonight, I have the Camera module installed. Can you send me some sample code for recording a video with react-native-camera? It will save me a little time.

**Comment by cobarx** [comment]
I was quite tired today and didn't feel up to coding. I should be able to do this tomorrow.

**Comment by jonathangreco** [comment]
@cobarx ping

Hi again, we've tested it again. Magically it works now.
I supposed this is indeed relative to the master branch. That's have not been tested well by my intern since the logcat output.

For now I will say its working on the master branch and not on the latest release, like you said.

My intern will try to validate this statement right now. Sorry for the inconvenience. And thanks to pointed that out.

**Comment by cobarx** [comment]
Great, glad this is sorted out. Btw, I have released 4.0.0 so you can now switch to that.
